version: '3.8'

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    image: ea-invitation-api
    container_name: ea-invitation-api
    restart: unless-stopped
    working_dir: /var/www
    volumes:
      - ./:/var/www
      - ./docker/php/local.ini:/usr/local/etc/php/conf.d/local.ini
    networks:
      - app-network

  webserver:
    image: nginx:alpine
    container_name: ea-invitation-webserver
    restart: unless-stopped
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./:/var/www
      - ./docker/nginx/conf.d/:/etc/nginx/conf.d/
      - ./docker/nginx/ssl:/etc/nginx/ssl
    networks:
      - app-network

  db:
    image: mysql:8.0
    container_name: ea-invitation-db
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: ea_invitation_db
      MYSQL_ROOT_PASSWORD: ea_root_password
      MYSQL_PASSWORD: ea_password
      MYSQL_USER: ea_user
    volumes:
      - dbdata:/var/lib/mysql
      - ./docker/mysql/my.cnf:/etc/mysql/my.cnf
    networks:
      - app-network
    ports:
      - "3306:3306"

networks:
  app-network:
    driver: bridge

volumes:
  dbdata:
    driver: local